service_grpc_influxrpc/src/service.rs

```diff
@@ -2922,6 +2924,232 @@ mod tests {
+                        predicate: Some(make_state_eq_ma_predicate()),
+                    };
+                    let streaming_resp = service
+                        .tag_keys(tonic::Request::new(request))
+                        .await
+                        .unwrap();
+                    Box::new(streaming_resp) as _
+                }
+                Self::TagValues => {
+                    let request = TagValuesRequest {
+                        tags_source: source.clone(),
+                        range: Some(make_timestamp_range(0, 2000)),
+                        predicate: Some(make_state_eq_ma_predicate()),
+                        tag_key: [255].into(),
+                    };
+                    let streaming_resp = service
+                        .tag_values(tonic::Request::new(request))
+                        .await
+                        .unwrap();
+                    Box::new(streaming_resp) as _
+                }
+                Self::TagValuesGroupedByMeasurementAndTagKey => {
+                    let request = TagValuesGroupedByMeasurementAndTagKeyRequest {
+                        source: source.clone(),
+                        measurement_patterns: vec![],
+                        tag_key_predicate: Some(TagKeyPredicate {
+                            value: Some(Value::Eq("state".into())),
+                        }),
+                        condition: None,
+                    };
+                    let streaming_resp = service
+                        .tag_values_grouped_by_measurement_and_tag_key(tonic::Request::new(request))
+                        .await
+                        .unwrap();
+                    Box::new(streaming_resp) as _
+                }
+            }
+        }
+
+        fn all() -> Vec<Self> {
+            vec![
+                Self::MeasurementFields,
+                Self::MeasurementNames,
+                Self::MeasurementTagKeys,
+                Self::MeasurementTagValues,
+                Self::ReadFilter,
+                Self::ReadGroup,
+                Self::ReadWindowAggregate,
+                Self::TagKeys,
+                Self::TagValues,
+                Self::TagValuesGroupedByMeasurementAndTagKey,
+            ]
+        }
+    }
+
+    #[tokio::test]
+    async fn test_query_semaphore() {
+        let semaphore_size = 2;
+        let test_storage = Arc::new(TestDatabaseStore::new_with_semaphore_size(semaphore_size));
+
+        // add some data
+        let db_info = org_and_bucket();
+        let chunk = TestChunk::new("TheMeasurement")
+            .with_time_column()
+            .with_tag_column("state")
+            .with_one_row_of_data();
+        test_storage
+            .db_or_create(db_info.db_name())
+            .await
+            .add_chunk("my_partition_key", Arc::new(chunk));
+
+        // construct request
+        for t in SemaphoredRequest::all() {
+            let service = StorageService {
```

Comment:
```suggestion
            println!("Testing with request: {:?}", t)
            let service = StorageService {
```

Maybe adding some logging would help in case the tests fails